pkg/controller/controller.go

```diff
@@ -605,3 +608,46 @@ func unsetLoggerCtxValue(ctx context.Context, key string) context.Context {
 
 	return ctx
 }
+
+func reportLastReconciled(o interface{}) {
+	var kind string
+	{
+		obj, ok := o.(pkgruntime.Object)
+		if !ok {
+			return
+		}
+
+		gvks, _, err := scheme.Scheme.ObjectKinds(obj)
+		if pkgruntime.IsNotRegisteredError(err) {
```

Comment:
If this object is not registered in k8s upstream runtime, probably we should use another runtime package in `apiextensions` repository.